unison-src/transcripts/tdnr.output.md

```diff
@@ -0,0 +1,1008 @@
+    
+      bad.foo  : Text
+      good.foo : Nat
+
+```
+\=== start local over direct dep
+
+TDNR selects local term (in file) that typechecks over direct dependency that doesn't.
+
+``` unison
+lib.bad.foo = "bar"
+```
+
+``` ucm
+
+  Loading changes detected in scratch.u.
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      lib.bad.foo : Text
+
+```
+``` ucm
+scratch/main> add
+
+  ⍟ I've added these definitions:
+  
+    lib.bad.foo : Text
+
+```
+``` unison
+good.foo = 17
+thing = foo Nat.+ foo
+```
+
+``` ucm
+
+  Loading changes detected in scratch.u.
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      good.foo : Nat
+      thing    : Nat
+
+```
+TDNR selects local term (in namespace) that typechecks over direct dependency that doesn't.
+
+``` unison
+good.foo = 17
+lib.bad.foo = "bar"
+```
+
+``` ucm
+
+  Loading changes detected in scratch.u.
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      good.foo    : Nat
+      lib.bad.foo : Text
+
+```
+``` ucm
+scratch/main> add
+
+  ⍟ I've added these definitions:
+  
+    good.foo    : Nat
+    lib.bad.foo : Text
+
+```
+``` unison
+thing = foo Nat.+ foo
+```
+
+``` ucm
+
+  Loading changes detected in scratch.u.
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      thing : Nat
+
+```
+TDNR selects local term (shadowing namespace) that typechecks over direct dependency that doesn't.
+
+``` unison
+good.foo = 17
+lib.bad.foo = "bar"
+```
+
+``` ucm
+
+  Loading changes detected in scratch.u.
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      good.foo    : Nat
+      lib.bad.foo : Text
+
+```
+``` ucm
+scratch/main> add
+
+  ⍟ I've added these definitions:
+  
+    good.foo    : Nat
+    lib.bad.foo : Text
+
+```
+``` unison
+good.foo = 18
+thing = foo Nat.+ foo
+```
+
+``` ucm
+
+  Loading changes detected in scratch.u.
+
+  I found and typechecked these definitions in scratch.u. If you
+  do an `add` or `update`, here's how your codebase would
+  change:
+  
+    ⍟ These new definitions are ok to `add`:
+    
+      thing : Nat
+    
+    ⍟ These names already exist. You can `update` them to your
+      new definition:
+    
+      good.foo : Nat
+
+```
+TDNR not used to select local term (in file) that typechecks over indirect dependency that also typechecks.
```

Comment:
Is this example _showing_ TDNR not being used? Or what is this example showing?

Reply:
Same with the next two "TDNR not used" examples

Reply:
Is it just informative, saying the local term (in file) is selected over indirect dependency, before TDNR happens? If so, maybe we should print the result again to demonstrate it.